### ISSUE TITLE
Builds are filled with warnings about false positives

### ISSUE THREAD

**Issue by harman28** [gocardless/gocardless-pro-ruby#37]
[A lot of tests](https://github.com/gocardless/gocardless-pro-ruby/search?utf8=%E2%9C%93&q=to_not+raise_error&type=) use the `expect { }.not_to raise_error(SpecificErrorClass)` construct. 

RSpec throws a warning for this, as the possibility of a false positive is high:
```
WARNING: Using `expect { }.not_to raise_error(SpecificErrorClass)` risks false 
positives, since literally any other error would cause the expectation to pass, 
including those raised by Ruby (e.g. NoMethodError, NameError and ArgumentError), 
meaning the code you are intending to test may not even get reached. Instead 
consider using `expect { }.not_to raise_error` or 
`expect { }.to raise_error(DifferentSpecificErrorClass)`. This message can be 
suppressed by setting: 
`RSpec::Expectations.configuration.on_potential_false_positives = :nothing`. 
```

You can see a lot of these warnings in [any build](https://travis-ci.org/gocardless/gocardless-pro-ruby/jobs/233978119).
I imagine the warning isn't applicable in a repo where you've got enough tests to cover all flows, but in that case it probably makes sense to set the right [config](https://www.rubydoc.info/github/rspec/rspec-expectations/RSpec%2FExpectations%2FConfiguration:on_potential_false_positives) to [suppress the warnings in spec_helper](https://github.com/harman28/gocardless-pro-ruby/commit/1ee985a6eb8b7a17857efe795a8a1b8d200af8f3#diff-93830fa29d616f7c87903d08b5b1b29aR32).

**Comment by harman28** [comment]
Also, I understand the intention is that developers will never have to download this code and run a build themselves, since this repo doesn't accept contributions. Is that why there's no `.gitignore` files and the generated `Gemfile.lock` shows up in my git diff?

```
harmansingh@rzp gocardless-pro-ruby (master) > gs
On branch master
Your branch is up to date with 'upstream/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	Gemfile.lock

nothing added to commit but untracked files present (use "git add" to track)
```

**Comment by nickcampbell18** [comment]
Hi @harman28, thanks for your questions!

1. We auto-generate the specs in this repo, hence you'll see the same RSpec errors all over the case. I've raised an internal ticket to stop checking for a particular error and instead just expect that no errors are raised.

2. We should have a `.gitignore` file since the `Gemfile.lock` will be generated whether we want it or not. I've raised another ticket for that too.
